### PR TITLE
Move to Go Modules

### PR DESCRIPTION
**Pull request by jen20**
This PR moves `pulumi-eks` to using Go Modules for managing Go dependencies.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
I am not qualified to review this :)  Maybe @ellismg .  All i care about is that this works smoothly :)

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,8 @@
+module github.com/pulumi/pulumi-eks
+
+go 1.12
+
+require (
+	github.com/pulumi/pulumi v0.17.8-0.20190418223219-0ede30fdb624
```
Comment (ellismg) [inline go.mod]: What is the workflow for updating this, @jen20?  I know I need to construct a pseudo version but is there a good way gesture in the toochain today for "pick up HEAD"?

Comment (jen20) [reply go.mod]: Yes - it's `GO111MODULE=on go get github.com/pulumi/pulumi@master` - the pseudo version will be determined automatically, and transitive dependencies will be updated. The `GO111MODULE=on` is required if the repository is inside `GOPATH` with Go 1.12.

Comment (ellismg) [reply go.mod]: :+1:

**Pull request merged**